compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -528,15 +528,15 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   def makeSelfDef(name: TermName, tpt: Tree)(using Context): ValDef =
     ValDef(name, tpt, EmptyTree).withFlags(PrivateLocal)
 
-  def makeTupleOrParens(ts: List[Tree])(using Context): Tree = ts match {
+  def makeTupleOrParens(ts: List[Tree])(using Context): Tree = ts match
+    case (t: NamedArg) :: Nil => Tuple(t :: Nil)
```

Comment:
will probably need a warning of changing behaviour, before this would be considered an assignment in parens:
```scala
scala> var x = 0
var x: Int = 0

scala> (x = 23)

scala> println(x)
23
```

Reply:
Noted for further work. We need to think of this migration when named tuples become standard.